src/rust/ide/view/graph-editor/src/component/visualization/container.rs

```diff
@@ -44,17 +44,35 @@ const CORNER_RADIUS : f32       = super::super::node::CORNER_RADIUS;
 pub mod background {
     use super::*;
 
-    // TODO use style
     ensogl::define_shape_system! {
-        (selected:f32,radius:f32,roundness:f32) {
+        (style:Style,selected:f32,radius:f32,roundness:f32) {
             let width  : Var<Pixels> = "input_size.x".into();
             let height : Var<Pixels> = "input_size.y".into();
-            let radius        = 1.px() * &radius;
-            let color_bg      = color::Lcha::new(0.2,0.013,0.18,1.0);
-            let corner_radius = &radius * &roundness;
-            let background    = Rect((&width,&height)).corners_radius(&corner_radius);
-            let background    = background.fill(color::Rgba::from(color_bg));
-            background.into()
+            let width             = &width  - SHADOW_SIZE.px() * 2.0;
+            let height            = &height - SHADOW_SIZE.px() * 2.0;
+            let radius            = 1.px() * &radius;
+            let color_bg_path     = "graph_editor.visualization.background.color";
+            let color_bg_fallback = color::Lcha::new(0.2,0.013,0.18,1.0);
+            let color_bg          = style.get_color_or(color_bg_path,color_bg_fallback);
+            let corner_radius     = &radius * &roundness;
+            let background        = Rect((&width,&height)).corners_radius(&corner_radius);
+            let background        = background.fill(color::Rgba::from(color_bg));
+
+            // === Shadow ===
+
+            let border_size_f = 16.0;
+            let corner_radius = corner_radius*1.75;
+            let width         = &width  + SHADOW_SIZE.px() * 2.0;
+            let height        = &height + SHADOW_SIZE.px() * 2.0;
+            let shadow        = Rect((&width,&height)).corners_radius(&corner_radius).shrink(1.px());
+            let shadow_color  = color::LinearGradient::new()
+                .add(0.0,color::Rgba::new(0.0,0.0,0.0,0.0).into_linear())
+                .add(1.0,color::Rgba::new(0.0,0.0,0.0,0.20).into_linear());
+            let shadow_color  = color::SdfSampler::new(shadow_color).max_distance(border_size_f).slope(color::Slope::Exponent(2.0));
```

Comment:
overlong

Reply:
Still overlong. Please do not mark conversations as resolved until you actually fix them! (or it will have a reply that it is agreed to leave it).